dlp_mpi/split/managed.py

```diff
@@ -105,6 +105,13 @@ def split_managed(
 
     failed_indices = []
 
+    # Clone acts as barrier and prevents the root process to receive wrong
+    # messages (i.e. when this function finishes on the worker they may can do
+    # again some communications).
```

Comment:
The i.e. part sounds strange. But I'm not sure how to formulate it. Maybe something like "i.e., when this function finishes on a worker, it might start some other communications, e.g., gathering results"

Reply:
Ok, I will try to find a better formulation. I think the gathering is not the right example, because gathering should also work without this fix. The problematic mpi command is "send".